loopy/symbolic.py

```diff
@@ -2000,7 +2000,7 @@ def simplify_using_aff(kernel, expr):
         cur_dim = domain.dim(isl.dim_type.set)
         domain = domain.insert_dims(isl.dim_type.set, cur_dim, len(non_inames))
         for non_iname in non_inames:
```

Comment:
Could the `non_inames` be sorted? Otherwise this might lead to non-deterministic codegen leading to memorization misses.